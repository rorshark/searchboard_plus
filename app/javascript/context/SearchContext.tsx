import React from 'react'
import { useSearch, SearchAPI, SearchParams } from '../hooks/useSearch'

export const defaultSearchApi: SearchAPI = {
  error: false,
  loading: true,
  customers: [],
  search: '',
  setSearch: (_) => {},
  companies: [],
  selectedCompany: null,
  setSelectedCompany: (_) => {},
  fetchCustomers: async () => {}
}

export const SearchContext: React.Context<SearchAPI> = React.createContext(defaultSearchApi)

export const SearchProvider: React.FC<{
  initialSearchParams: SearchParams
}> = ({ initialSearchParams, children }) => {
  const search = useSearch(initialSearchParams)

  return (
    <SearchContext.Provider value={search}>
      {children}
    </SearchContext.Provider>
  )
}
