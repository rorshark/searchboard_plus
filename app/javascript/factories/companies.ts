import { Factory } from 'fishery'
import Faker from 'faker'
import { Company } from '../types/companies'

export const company = Factory.define<Company>(() => ({
  companyName: Faker.company.companyName()
}))
