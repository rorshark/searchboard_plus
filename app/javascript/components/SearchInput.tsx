import React, { useContext } from 'react'
import { SearchContext } from '../context/SearchContext'
import { TextField } from '@material-ui/core'

const SearchInput: React.FC = () => {
  const { search, setSearch } = useContext(SearchContext)

  return (
    <TextField
      label="Search" 
      variant="outlined"
      value={search}
      onChange={(evt) => setSearch(evt.target.value) }
      inputProps={{
        "data-testid": "search-input"
      }}
    />
  )
}

export default SearchInput
