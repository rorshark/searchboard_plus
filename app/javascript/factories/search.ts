import { Factory } from 'fishery'
import { SearchAPI, SearchParams } from '../hooks/useSearch'
import { defaultSearchApi } from '../context/SearchContext'

export const searchApi = Factory.define<SearchAPI>(() => defaultSearchApi)

export const searchParams = Factory.define<SearchParams>(() => ({
  search: '',
  filter_by_company_name: ''
}))
