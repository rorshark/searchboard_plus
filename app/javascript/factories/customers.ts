import { Factory } from 'fishery'
import Faker from 'faker'
import { Customer } from '../types/customers'
import { company } from './companies'

export const customer = Factory.define<Customer>(() => ({
  firstName: Faker.name.firstName(),
  lastName: Faker.name.lastName(),
  company: company.build()
}));
