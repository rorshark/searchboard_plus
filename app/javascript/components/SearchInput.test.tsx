import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { SearchContext } from '../context/SearchContext'
import { searchApi } from '../factories/search'
import SearchInput from './SearchInput'

test('it renders a text input', () => {
  const context = searchApi.build()

  render(
    <SearchContext.Provider value={context}>
      <SearchInput />
    </SearchContext.Provider>
  )

  const element = screen.getByTestId('search-input')

  expect(element).toMatchSnapshot()
})

test('it populates the input field', () => {
  const context = searchApi.build({
    search: 'Dougy Pauly'
  })

  render(
    <SearchContext.Provider value={context}>
      <SearchInput />
    </SearchContext.Provider>
  )

  const element = screen.getByTestId('search-input')

  expect(element).toMatchSnapshot()
})

test('it calls setSearch', () => {
  const setSearch = jest.fn()

  const context = searchApi.build({
    search: 'Dougy Pauly',
    setSearch
  })

  render(
    <SearchContext.Provider value={context}>
      <SearchInput />
    </SearchContext.Provider>
  )

  const element = screen.getByTestId('search-input')

  fireEvent.change(element, { target: { value: 'Test' } })

  expect(setSearch.mock.calls[0][0]).toEqual('Test')
})
