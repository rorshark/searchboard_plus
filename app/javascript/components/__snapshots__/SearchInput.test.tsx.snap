// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it populates the input field 1`] = `
<input
  data-testid="search-input"
  name="search"
  type="text"
  value="Dougy Pauly"
/>
`;

exports[`it renders a text input 1`] = `
<input
  data-testid="search-input"
  name="search"
  type="text"
  value=""
/>
`;
